### PR TITLE
[MM-30372] Cypress/E2E: Automate backlogs - Messaging > Reply

### PR DESCRIPTION
**Pull request by streamer45**
#### Summary

PR implements:

- [MM-T2132](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T2132)
- [MM-T2133](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T2133)
- [MM-T2134](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T2134)
- [MM-T2135](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T2135)
- [MM-T2136](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T2136)

#### Ticket

https://mattermost.atlassian.net/browse/MM-30372

### CODE REVIEW

File: e2e/cypress/integration/messaging/message_reply_part2_spec.js [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    let otherChannel;
+    let rootId;
+
+    before(() => {
+        // # Create main channel
+        cy.apiInitSetup({loginAfter: true}).then(({team, channel}) => {
+            mainChannel = channel;
+
+            // # Create other channel
+            cy.apiCreateChannel(team.id, 'other', 'other').then(({newChannel}) => {
```
Comment (saturninoabril) [inline e2e/cypress/integration/messaging/message_reply_part2_spec.js:25]: MM-T2136 is failing on me. I believe this should be,
```suggestion
            cy.apiCreateChannel(team.id, 'other', 'other').then(({channel: newChannel}) => {
```

**Review by saturninoabril — CHANGES_REQUESTED** [review]
Thanks @streamer45. LGTM, except for one comment.

File: e2e/cypress/integration/messaging/message_reply_part2_spec.js (hunk 1 above)
Comment (streamer45) [reply e2e/cypress/integration/messaging/message_reply_part2_spec.js:25]: oh yeah, didn't notice it was destructuring the object, thanks :+1:

**Review by saturninoabril — APPROVED** [review]
Thanks @streamer45!

File: e2e/cypress/integration/messaging/message_reply_part2_spec.js [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+        // # Close RHS
+        cy.closeRHS();
+    });
+
+    it('MM-T2136 - Reply in RHS with different channel open in center', () => {
+        // # Click on the `...` menu icon and click on `Reply`
+        cy.uiClickPostDropdownMenu(rootId, 'Reply', 'CENTER');
+
+        // # Switch to a different channel
+        cy.get(`#sidebarItem_${otherChannel.name}`).click().wait(TIMEOUTS.FIVE_SEC);
```
Comment (isacikgoz) [inline e2e/cypress/integration/messaging/message_reply_part2_spec.js:136]: Does this fail under 5 seconds? It seemed quite long to me 😅

**Review by isacikgoz — APPROVED** [review]
LGTM 👍

File: e2e/cypress/integration/messaging/message_reply_part2_spec.js (hunk 2 above)
Comment (streamer45) [reply e2e/cypress/integration/messaging/message_reply_part2_spec.js:136]: Not sure but I've seen it used when switching channels on other tests and ported it here. It's to make sure all state has been properly loaded.

**Review by lindy65 — APPROVED** [review]
Thanks @streamer45 - looks good to me 👍

**Pull request merged**